File: DZ5_DZ6/ShopService/Shipment.java
package DZ5_DZ6.ShopService;

import java.util.ArrayList;

import DZ5_DZ6.Phone.Phone;

public class Shipment implements IntrfTransaction {

    @Override
    public void changingTheBalance(Phone phone, PhoneMag mag) {
        ArrayList<Phone> priceList = new ArrayList<>();
        priceList = mag.getPriceList();
        if (mag.getMoney() >= phone.getPrice()){
            mag.setMoney(mag.getMoney()-phone.getPrice());

            phone.setPrice(phone.getPrice()*1.1);
            priceList.add(phone);
            mag.setPriceList(priceList);
        }
        else{
            PrintMassage.printMassage("Недостаточно средств!");
        }
    }

    
}

File: DZ7/ShopService/Shipment.py

from Phone.Phone import Phone
from ShopService.PhoneMag import PhoneMag
from ShopService.PrintMassage import PrintMassage


class Shipment:
    @staticmethod
    def changingTheBalance( phone:Phone,  mag:PhoneMag):
        priceList = mag.getPriceList()
        if mag.getMoney() >= phone.getPrice():
            mag.setMoney(mag.getMoney()-phone.getPrice())

            phone.setPrice(phone.getPrice()*1.1)
            priceList.append(phone)
            mag.setPriceList(priceList)
        else:
            PrintMassage.printMassage("Недостаточно средств!")
